test/src/main/java/org/apache/accumulo/test/functional/ThriftMaxFrameSizeIT.java

```diff
@@ -56,50 +62,95 @@ class TestDefault extends TestMaxFrameSize {
+    }
   }
 
   @Nested
   class TestThreadedSelector extends TestMaxFrameSize {
     TestThreadedSelector() {
       serverType = ThriftServerType.THREADED_SELECTOR;
     }
+
+    @Test
+    public void testThreadedSelectorServerFrameSize() throws Exception {
+      AtomicBoolean succeeded = new AtomicBoolean(false);
+      assertThrows(AssertionFailedError.class,
+          () -> assertTimeoutPreemptively(Duration.ofSeconds(30),
+              () -> testMaxFrameSizeLargerThanDefault(succeeded)));
+      assertFalse(succeeded.get());
+    }
   }
 
   @Nested
   class TestCustomHsHa extends TestMaxFrameSize {
     TestCustomHsHa() {
       serverType = ThriftServerType.CUSTOM_HS_HA;
     }
+
+    @Test
+    public void testCustomHsHaServerFrameSize() throws Exception {
+      AtomicBoolean succeeded = new AtomicBoolean(false);
+      assertThrows(AssertionFailedError.class,
+          () -> assertTimeoutPreemptively(Duration.ofSeconds(30),
+              () -> testMaxFrameSizeLargerThanDefault(succeeded)));
+      assertFalse(succeeded.get());
+    }
   }
 
   @Nested
   class TestThreadPool extends TestMaxFrameSize {
     TestThreadPool() {
       serverType = ThriftServerType.THREADPOOL;
     }
+
+    @Test
+    public void testThreadPoolServerFrameSize() throws Exception {
+      AtomicBoolean succeeded = new AtomicBoolean(false);
+      assertThrows(AssertionFailedError.class,
+          () -> assertTimeoutPreemptively(Duration.ofSeconds(30),
+              () -> testMaxFrameSizeLargerThanDefault(succeeded)));
+      assertFalse(succeeded.get());
+    }
   }
 
   @Nested
   class TestSsl extends TestMaxFrameSize {
     TestSsl() {
-      serverType = ThriftServerType.THREADPOOL;
+      serverType = ThriftServerType.SSL;
+    }
+
+    @Test
+    public void testSslServerFrameSize() throws Exception {
+      AtomicBoolean succeeded = new AtomicBoolean(false);
+      testMaxFrameSizeLargerThanDefault(succeeded);
+      assertTrue(succeeded.get());
     }
   }
 
   protected abstract class TestMaxFrameSize {
 
-    @Test
-    public void testMaxFrameSizeLargerThanDefault() throws Exception {
+    public void testMaxFrameSizeLargerThanDefault(AtomicBoolean success) throws Exception {
+
+      int maxSize = TConfiguration.DEFAULT_MAX_FRAME_SIZE;
+      // make sure we go even bigger than that
+      int ourSize = maxSize * 2;
 
       // Ingest with a value width greater than the thrift default size to verify our setting works
-      // for max frame wize
+      // for max frame size
       try (AccumuloClient accumuloClient = Accumulo.newClient().from(getClientProps()).build()) {
-        String table = getUniqueNames(1)[0];
-        ReadWriteIT.ingest(accumuloClient, 1, 1, TConfiguration.DEFAULT_MAX_FRAME_SIZE + 1, 0,
-            table);
-        ReadWriteIT.verify(accumuloClient, 1, 1, TConfiguration.DEFAULT_MAX_FRAME_SIZE + 1, 0,
-            table);
+        String table = getUniqueNames(1)[0] + serverType.name();
```

Comment:
I am not sure why this was modified to add the server type name to the table name. If the `@Nested` naming is working correctly, these already have unique names per test.

Reply:
As it turns out, the naming wasn't unique. It's actually worse than that, though... minicluster reuses directories from other tests, because the test naming for nested classes are not unique enough for it to use a separate test directory. I'm not exactly sure how to fix that, as I'm not super familiar with the nested test concept, but as long as the test is passing, and they aren't being run concurrently, it's not really a big deal.